cli/azd/pkg/azdo/utils.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+)
+
+// helper method to verify that a configuration exists in the .env file or in system environment variables
+func ensureConfigExists(ctx context.Context, env *environment.Environment, key string, label string) (string, error) {
+	value := env.Values[key]
+	if value != "" {
+		return value, nil
+	}
+
+	value, exists := os.LookupEnv(key)
+	if !exists || value == "" {
+		return value, fmt.Errorf("%s not found in environment variable %s", label, key)
+	}
+	return value, nil
+}
+
+// helper method to ensure an Azure DevOps PAT exists either in .env or system environment variables
+func EnsurePatExists(ctx context.Context, env *environment.Environment, console input.Console) (string, error) {
+	value, err := ensureConfigExists(ctx, env, AzDoPatName, "azure devops personal access token")
+	if err != nil {
```

Comment:
Consider inverting this if statement to reduce nesting and do a short circuit in the case where you have a value and no error.

Reply:
https://github.com/Azure/azure-dev/issues/756